src/QAT/qat/purr/integrations/tket.py

```diff
@@ -366,37 +389,33 @@ def run_tket_optimizations(qasm_string, opts, hardware: QuantumHardwareModel) ->
         )
         return qasm_string
 
-    couplings = hardware.qubit_direction_couplings
+    couplings = deepcopy(hardware.qubit_direction_couplings)
```

Comment:
why does this need deep copying?

Reply:
It's because we do not want to actually change the couplings on the hardware model.